backends/magma/ceed-magma-det.c

```diff
@@ -19,40 +19,17 @@ static int CeedInit_Magma_Det(const char *resource, Ceed ceed) {
   CeedCallBackend(CeedSetDeterministic(ceed, true));
 
   Ceed_Magma *data;
-  CeedCallBackend(CeedCalloc(sizeof(Ceed_Magma), &data));
+  CeedCallBackend(CeedCalloc(1, &data));
   CeedCallBackend(CeedSetData(ceed, data));
 
-  // get/set device ID
-  const char *device_spec = strstr(resource, ":device_id=");
-  const int   deviceID    = (device_spec) ? atoi(device_spec + 11) : -1;
-
-  int currentDeviceID;
-  magma_getdevice(&currentDeviceID);
-  if (deviceID >= 0 && currentDeviceID != deviceID) {
-    magma_setdevice(deviceID);
-    currentDeviceID = deviceID;
-  }
-  // create a queue that uses the null stream
-  data->device = currentDeviceID;
-
```

Comment:
Since you mentioned this in your TODO:  I do think we still need this here, for the case where we want to set a device ID other than default (0).  Not only do we want to set the device in the backend data (line 36), but we want to make sure the `magma` backend `ceedref` objects uses the correct device.  If we've already set the current device to something other than 0, then the `magma` delegate object will use the current value (after calling `magma_getdevice` in its own initialization), even though we don't explicitly pass the device ID in the resource string when creating `ceedref` (line 42 or 44).  It doesn't have to be this exact code, but we do need to make sure we can handle non-zero device ID specifications.

Reply:
Ah, thanks @nbeams! I've added this back in, via a common `ceed-magma-common.h/c` similar to what is done for `ceed-cuda/hip-common.h/c`. This should now be resolved!